src/types/index.ts

```diff
@@ -84,21 +84,44 @@ type UserTransformationLibrary = {
   VersionID: string;
 };
 
-type ProcessorTransformationRequest = {
-  request?: object;
-  message: object;
-  metadata: Metadata;
-  destination: Destination;
-  libraries: UserTransformationLibrary[];
-};
+type RudderTimestamp = Date | string;
+type RudderMessageType =
+  | 'identify'
+  | 'track'
+  | 'page'
+  | 'screen'
+  | 'group'
+  | 'alias'
+  | 'audiencelist';
+
+type RudderMessageSchema = Partial<{
+  userId: string;
+  anonymousId: string;
+  type: RudderMessageType;
+  channel: string;
+  context: object;
+  originalTimestamp: RudderTimestamp;
+  sentAt: RudderTimestamp;
+  timestamp: RudderTimestamp;
+  event: string;
+  integrations: object;
+  messageId: string;
+  properties: object;
+  traits: object;
+  [key: string]: FixMe;
+}>;
 
 type RouterTransformationRequestData = {
```

Comment:
This is low level struct used in both process & routerTransform flow, so shouldn't the name be neutral ?
```suggestion
type EventRequestData = {
```

Reply:
Basically it was named as `RouterTransformationRequestData` previously but since the elements in the type were common, I used it as a low-level struct.

Tried to change name, there are many files which are using this struct. I feel this is a little bigger change & out of scope of this task. May be we can handle this in a backlog item and make a change independent of this task. But yeah, let me know if we are ok with making this change as part of this PR.